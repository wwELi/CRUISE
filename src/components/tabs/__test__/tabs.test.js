
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Tabs from '../index';

Enzyme.configure({ adapter: new Adapter() });

const tabs = [
  { key: 'all', name: 'All' },
  { key: 'virtual', name: 'Virtual' },
  { key: 'physical', name: 'Physical' }
];


describe('tabs', () => {
  it('renders', () => {

    const onChangeMock = () => {};


    const tree = renderer
      .create(<Tabs tabs={tabs} onTabChange={onChangeMock}>hello</Tabs>)
      .toJSON();
    expect(tree).toMatchSnapshot();

  });

  describe('onChange', () => {

    let wrapper;
    let handle;

    beforeEach(() => {
      handle = jest.fn();
      wrapper = mount(
        <Tabs tabs={tabs} onTabChange={() => {}}>hello</Tabs>,
      );
    });

    it('choose length', () => {

      const wrapper = render(<Tabs tabs={tabs} defaultKey={'all'} onTabChange={() => {}}>hello</Tabs>);

      expect(wrapper.find('.tab-choose').length).toBe(1);
    })

  })

});

