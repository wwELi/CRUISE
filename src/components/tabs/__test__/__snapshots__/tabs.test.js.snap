// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabs renders 1`] = `
<div
  className="cm-tabs"
>
  <span
    className=""
    onClick={[Function]}
  >
    All
  </span>
  <span
    className=""
    onClick={[Function]}
  >
    Virtual
  </span>
  <span
    className=""
    onClick={[Function]}
  >
    Physical
  </span>
</div>
`;
